base/inference.jl

```diff
@@ -667,32 +707,32 @@ function apply_type_tfunc(args...)
     elseif isa(headtype, Union)
         return Any
     end
-    istuple = (headtype === Tuple)
+    istuple = (headtype == Tuple)
     uncertain = false
     tparams = Any[]
     for i=2:largs
         ai = args[i]
         if isType(ai)
             aip1 = ai.parameters[1]
-            uncertain |= has_typevars(aip1)
+            uncertain |= has_free_typevars(aip1)
             push!(tparams, aip1)
         elseif isa(ai, Const) && valid_tparam(ai.val)
             push!(tparams, ai.val)
         else
-            if !istuple && i-1 > length(headtype.parameters)
-                # too many parameters for type
-                return Bottom
-            end
+            #if !istuple && i-1 > length(headtype.parameters)
+            #    # too many parameters for type
+            #    return Bottom
+            #end
```

Comment:
This is tricky, since this error now happens only when trying to apply a non-UnionAll type. Fortunately this is a very uncommon case.